torch/_dynamo/profiler.py

```diff
@@ -15,21 +15,21 @@ class ProfileMetrics:
     fusions: int = 0
     graphs: int = 0
 
-    def __iadd__(self, other: "ProfileMetrics"):
+    def __iadd__(self, other: "ProfileMetrics") -> Self:
         self.microseconds += other.microseconds
         self.operators += other.operators
         self.fusions += other.fusions
         return self
 
-    def __add__(self, other: "ProfileMetrics"):
+    def __add__(self, other: "ProfileMetrics") -> "ProfileMetrics":
```

Comment:
Not sure about where else in this file Self is valid, but feel free to give it a try. :)

Reply:
Alas this doesn't work

```
  Error (MYPY) [return-value]
    Incompatible return value type (got "ProfileMetrics", expected "Self")

         32  |    def __truediv__(self, other: Any) -> Self:
         33  |        if isinstance(other, int):
         34  |            other = ProfileMetrics(other, other, other)
    >>>  35  |        return ProfileMetrics(
         36  |            self.microseconds / max(1, other.microseconds),
         37  |            self.operators / max(1, other.operators),
         38  |            self.fusions / max(1, other.fusions),
```